lib/ace/mode/lucene_highlight_rules.js

```diff
@@ -9,35 +9,41 @@ var LuceneHighlightRules = function() {
     this.$rules = {
         "start" : [
             {
-                token : "constant.character.negation",
-                regex : "[\\-]"
+                token: "constant.character.negation",
+                regex: "\\-"
             }, {
-                token : "constant.character.interro",
-                regex : "[\\?]"
+                token: "constant.character.interro",
+                regex: "\\?"
             }, {
-                token : "constant.character.asterisk",
-                regex : "[\\*]"
+                token: "constant.character.required",
+                regex: "\\+"
+            }, {
+                token: "constant.character.asterisk",
+                regex: "\\*"
             }, {
                 token: 'constant.character.proximity',
-                regex: '~[0-9]+\\b'
+                regex: '~(?:0\\.[0-9]+|[0-9]+)?'
+            }, {
+                token: 'keyword.operator',
+                regex: '(AND|OR|NOT|TO)\\b'
             }, {
-                token : 'keyword.operator',
-                regex: '(?:AND|OR|NOT)\\b'
+                token: "paren.lparen",
+                regex: "[\\(\\{\\[]"
             }, {
-                token : "paren.lparen",
-                regex : "[\\(]"
+                token: "paren.rparen",
+                regex: "[\\)\\}\\]]"
             }, {
-                token : "paren.rparen",
-                regex : "[\\)]"
+                token: "keyword",
+                regex: "\\S+:"
             }, {
-                token : "keyword",
-                regex : "[\\S]+:"
+                token: "string",           // " string
+                regex: '"[^"]*"'
```

Comment:
is there a way to escape characters in strings (\n \t \"), and can strings be multiline?

Reply:
as far as i can tell, this syntax is never intended to be used across multiple lines. the docs do mention escape sequences though, so i should change this to account for `\"` i think